src/Forms/Main.cs

```diff
@@ -17166,66 +17167,91 @@ private void SetShortcuts()
+                Version = Convert.ToDecimal(t.GetProperty(nameof(PluginInfoLocal.Version))?.GetValue(pluginObject, null)),
+                Author = (string)t.GetProperty(nameof(PluginInfoLocal.Author))?.GetValue(pluginObject, null),
+                Shortcut = (string)t.GetProperty(nameof(PluginInfoLocal.Shortcut))?.GetValue(pluginObject, null),
+                File = pluginFileName
+            };
 
-                pi = t.GetProperty("Text");
-                if (pi != null)
-                    text = (string)pi.GetValue(pluginObject, null);
+            // File, Tool, Sync, Translate, Spellcheck
+            var pi = t.GetProperty("ActionType");
+            if (pi != null)
+            {
+                string at = (string)pi.GetValue(pluginObject, null);
+                at = at.CapitalizeFirstLetter(); 
+                ActionType actionType = (ActionType)Enum.Parse(typeof(ActionType), at);
+                pluginInfo.ActioType = actionType;
+            }
 
-                pi = t.GetProperty("Description");
-                if (pi != null)
-                    description = (string)pi.GetValue(pluginObject, null);
+            var mi = t.GetMethod("DoAction");
+            if (mi == null)
+            {
+                return null;
+            }
 
-                pi = t.GetProperty("Version");
-                if (pi != null)
-                    version = Convert.ToDecimal(pi.GetValue(pluginObject, null));
+            // object instance wich will be used to invoke method
+            var instance = Expression.Constant(pluginObject);
 
-                pi = t.GetProperty("ActionType");
-                if (pi != null)
-                    actionType = (string)pi.GetValue(pluginObject, null);
+            // create plugin DoAction parameters
+            var formParam = Expression.Parameter(typeof(Form), "parentForm");
+            var srtTextParm = Expression.Parameter(typeof(string), "srtText");
+            var frameRateParam = Expression.Parameter(typeof(double), "frameRate");
+            var uiLineBreakParam = Expression.Parameter(typeof(string), "uiLineBreak");
+            var fileParam = Expression.Parameter(typeof(string), "file");
+            var videoFileParam = Expression.Parameter(typeof(string), "videoFile");
+            var rawFileParam = Expression.Parameter(typeof(string), "rawText");
 
-                mi = t.GetMethod("DoAction");
+            // create method call expression
+            var methodCallExp = Expression.Call(instance, mi, formParam, srtTextParm,
+                frameRateParam, uiLineBreakParam, fileParam, videoFileParam, rawFileParam);
 
-                pi = t.GetProperty("Shortcut");
-                if (pi != null)
-                    shortcut = (string)pi.GetValue(pluginObject, null);
+            // make lambda from method call expression
+            var doAction = Expression.Lambda<Func<Form, string, double, string,

```

Comment:
Constructed only once, so every time a plugin is invoked it will invoke the pre-constructed delegate directly